File: processor/src/test/resources/scenarios/cassandracodecs/AnotherValidCassandraCodec.java
package scenarios.cassandracodecs;

import com.datastax.oss.driver.api.core.type.codec.MappingCodec;
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs;
import com.datastax.oss.driver.api.core.type.reflect.GenericType;

public class AnotherValidCassandraCodec extends MappingCodec<Integer, Integer> {

    public AnotherValidCassandraCodec() {
        super(TypeCodecs.INT, GenericType.of(Integer.class));
    }

    @Override
    protected Integer innerToOuter(Integer value) {
        return value;
    }

    @Override
    protected Integer outerToInner(Integer value) {
        return value;
    }
}

File: ksp/src/test/resources/scenarios/cassandracodecs/AnotherValidCassandraCodec.kt
package scenarios.cassandracodecs

import com.datastax.oss.driver.api.core.type.codec.MappingCodec
import com.datastax.oss.driver.api.core.type.codec.TypeCodecs
import com.datastax.oss.driver.api.core.type.reflect.GenericType

class AnotherValidCassandraCodec : MappingCodec<Int, Int>(TypeCodecs.INT, GenericType.of(Int::class.javaObjectType)) {
    override fun innerToOuter(value: Int?): Int? = value
    override fun outerToInner(value: Int?): Int? = value
}
